### PR TITLE
Enhancement/gcp fix errors

### PR DESCRIPTION
**Pull request by SophieDorval**
Fix errors in GCP KMS keys

**Make sure the PR is against the `develop` branch (see [Contributing](https://github.com/nccgroup/ScoutSuite/blob/master/CONTRIBUTING.md)).**

Please include a summary of the change(s) and which issue(s) it addresses. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Select the relevant option(s):

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/providers/gcp/resources/kms/keys.py [hunk 1]
```diff
@@ -42,10 +42,13 @@ def _parse_key(self, raw_key):
...
+            rotation_period = int("".join(filter(str.isdigit, key_dict['rotation_period'])))
+            # check if rotation period less than 90 days or 7776000 seconds
+            key_dict['rotation_period'] = rotation_period <= 7776000
 
         key_dict['next_rotation_time_days'] = None
         if key_dict['next_rotation_datetime']:
             next_rotation_time = dateutil.parser.parse(key_dict['next_rotation_datetime']) - datetime.now(timezone.utc)
-            key_dict['next_rotation_time_days'] = next_rotation_time.days
+            # check if next rotation period less than 90 days
+            key_dict['next_rotation_time_days'] = next_rotation_time.days <= 90
```
Comment (j4v) [inline ScoutSuite/providers/gcp/resources/kms/keys.py:53]: How is this being displayed in the UI? You're moving the logic of the rule to the parsing but do these 2 variables still make sense considering what they are now storing?

File: ScoutSuite/providers/gcp/rules/findings/kms-encryption-keys-not-rotated.json [hunk 2]
```diff
@@ -35,8 +35,8 @@
             "or",
             [
                 "kms.projects.id.keyrings.id.keys.id.next_rotation_time_days",
-                "greaterThan",
-                "90"
+                "false",
```
Comment (j4v) [inline ScoutSuite/providers/gcp/rules/findings/kms-encryption-keys-not-rotated.json:38]: In terms of logic the previous implementation made more sense - you're no longer storing rotation information, only the result of the check.

File: ScoutSuite/providers/gcp/resources/kms/keys.py (hunk 1 above)
Comment (SophieDorval) [reply ScoutSuite/providers/gcp/resources/kms/keys.py]: Fix it.

File: ScoutSuite/providers/gcp/rules/findings/kms-encryption-keys-not-rotated.json (hunk 2 above)
Comment (SophieDorval) [reply ScoutSuite/providers/gcp/rules/findings/kms-encryption-keys-not-rotated.json]: Went back to the previous implementation

**Review by j4v — APPROVED** [review]

**Pull request merged**